### PR TITLE
update coding challenge requirements

### PR DESCRIPTION
**Pull request by w01fgang**

### CODE REVIEW

File: src/theme.ts [hunk 1]
```diff
@@ -3,65 +3,72 @@ const theme = {
     primary: {
       main: '#FF5500',
       light: '#FD6E35',
-      dark: '#CC4400'
+      dark: '#CC4400',
```
Comment (w01fgang) [inline src/theme.ts:6]: hm... it seems that vscode added all these commas 🤔

File: CHALLENGE.md [hunk 2]
```diff
@@ -94,9 +95,35 @@ You are given a prepared SPA project that was bootstrapped with Create React App
 ![create-tournament-prompt](.github/create-tournament-prompt.png)
 
 - Clicking on the `CREATE TOURNAMENT` button will open a browser prompt with a message `Tournament Name:`, an input field to enter the tournament name and the buttons `Cancel` to cancel and `OK` to confirm.
-- When confirming, the tournament will be created on the fake REST API and added on the UI without any loading indicators.
+- The tournament name must contain only Latin letters, numbers, and spaces, not an empty string or only spaces.
+- When confirming, the tournament will be created immediately using an optimistic update in the UI and a fetch call on the fake REST API.
```
Comment (chornaya-com) [inline CHALLENGE.md:99]: if I'm not mistaken we don't expect the candidates to implement optimistic update, when new tournament is created, only when edited or deleted.

File: CHALLENGE.md [hunk 3]
```diff
@@ -94,9 +95,35 @@ You are given a prepared SPA project that was bootstrapped with Create React App
...
+  <img alt="iMac Retina 27"" src=".github/iMac-Retina-27.png">
+</details>
 
 ## Finished?
 
-- Create a `ZIP` file with your solution
-- Send it to the recruiter that sent you this challenge
+- Create a `ZIP` file with your solution.
+- Include the .git folder to the `ZIP` file to make it easier for us to review your code.
+- Send it to the recruiter that sent you this challenge.
```
Comment (chornaya-com) [inline CHALLENGE.md:129]: should we also ask not to include node modules?
we might get back to repos, so hopefully this section will be changed

**Review by chornaya-com — APPROVED** [review]
LGTM 👍 Thanks for your job!
Could you, please, hold on from merging it, when it will be reviewed and approved by someone else, until we sync it with HR team, so current candidates who already implement this challenge won't be affected with this updates.

File: CHALLENGE.md [hunk 4]
```diff
@@ -52,14 +52,15 @@ You are given a prepared SPA project that was bootstrapped with Create React App
 ![edit-promp](.github/edit-prompt.png)
 
 - Clicking on the `EDIT` button will open a browser prompt with a message `New Tournament Name:`, an input field with the current tournament name as pre-filled value and the buttons `Cancel` to cancel and `OK` to confirm.
-- When confirming, the tournament name will be updated on the fake REST API and UI optimistically without any loading indicators.
+- The tournament name must contain only Latin letters, numbers, and spaces, not an empty string or only spaces.
+- When confirming, the tournament name will be updated immediately using an optimistic update in the UI and a fetch call on the fake REST API.
```
Comment (stefanosusini) [inline CHALLENGE.md:56]: Not sure what we said in our discussion. Shall we mention handling the error cases?

**Review by stefanosusini — APPROVED** [review]

File: CHALLENGE.md (hunk 3 above)
Comment (w01fgang) [reply CHALLENGE.md:129]: yes, I thought to mention it, but this could be a sign that the candidate doesn't know how npm modules work 😬
I'll add it, thanks!

File: CHALLENGE.md (hunk 4 above)
Comment (w01fgang) [reply CHALLENGE.md:56]: I would mention it even if we didn't discuss it.

File: CHALLENGE.md (hunk 2 above)
Comment (w01fgang) [reply CHALLENGE.md:99]: then it's not consistent
also, it's the same logic, I don't believe that will take more than 5 minutes to implement if you already have optimistic updates for edit/delete in place.

File: CHALLENGE.md (hunk 3 above)
Comment (chornaya-com) [reply CHALLENGE.md:129]: Here are more details on that https://faceit.slack.com/archives/GMSMHQAQJ/p1657551725576399

File: CHALLENGE.md (hunk 2 above)
Comment (chornaya-com) [reply CHALLENGE.md:99]: Here are more details on that https://faceit.slack.com/archives/GMSMHQAQJ/p1657551725576399 with the point from Michele, why we don't and should not ask about it

Comment (RicardoRodrigues) [reply CHALLENGE.md:99]: I understand could be not consistence , but could lead other problems, such the id not exist (then using different key when using map), or add a template data. Will keep simple only Edit and Delete, and continue how it was.

**Review by alumedot — APPROVED** [review]
Great job, @w01fgang 👍

File: CHALLENGE.md [hunk 5]
```diff
@@ -94,9 +92,43 @@ You are given a prepared SPA project that was bootstrapped with Create React App
...
+- The tournament name must contain only Latin letters, numbers, and spaces, not an empty string or only spaces.
+- When confirming, the tournament will be created immediately using an optimistic update in the UI and a fetch call on the fake REST API.
+
+### 6. Style it
+
+- The tournament has a border radius of `4px`.
+- The tournament name has the heading size `h6`.
+- The `Start` date must be displayed in the format `DD/MM/YYYY, HH:mm:ss` using the `en-GB` locale.
+- The horizontal spacing between the `EDIT` and `DELETE` button is `8px`.
+- The horizontal and vertical spacing between each tournament must be `24px`.
```
Comment (rb2610) [inline CHALLENGE.md:104]: Should we specify other spacings, for example between Tournament name and details, between details and buttons, etc?

It seemed a bit inconsistent for us to specify a few spacings but not others

File: CHALLENGE.md [hunk 6]
```diff
@@ -2,10 +2,10 @@
 
 ## Prerequisites
 
-- Clone this repository to your machine
-- Select the branch (depending on the language you want to work with)
-  - The `master` branch is based on TypeScript.
-
+- Clone this repository to your machine.
+- Please, avoid changing the repository structure without a strict necessity; otherwise, motivate your choice.
```
[the remark this answers is not in the document]
Comment (w01fgang) [reply CHALLENGE.md:6]: Well, I'm not a native English speaker, if it looks weird, then feel free to rephrase it.

File: CHALLENGE.md (hunk 5 above)
Comment (w01fgang) [reply CHALLENGE.md:104]: We try to find golden mean between being specific and overwhelming with requirements, if understand correctly.

File: CHALLENGE.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (i-oliva) [reply CHALLENGE.md:6]: I would rephrase it to 

Do not change the repository structure. If you change it, let us know why in the README.

File: CHALLENGE.md [hunk 7]
```diff
@@ -2,22 +2,23 @@
...
+- Write the code you are usually writing. Avoid writing broken or sketchy code. We can't evaluate how good you are if the code is not good.
```
Comment (i-oliva) [inline CHALLENGE.md:7]: :thinking: is this needed at all? I don't think we should warn the candidate to "please write good code 😅".

File: CHALLENGE.md [hunk 8]
```diff
@@ -2,22 +2,23 @@
...
 ![preview](.github/preview.gif)
 
 ### Notes
 
+- Challenge _must_ be implemented on TypeScript.
 - Redux _must_ be used for data fetching.
 - Alternatively to `styled-components`, you may use [CSS/CSS Modules/Sass provided by CRA](https://create-react-app.dev/docs/adding-a-stylesheet).
 - Any _open-source_ library is allowed to be used.
-- It should take around 2 hours to complete, max 3.
+- It should take around 3 hours to complete.
```
Comment (i-oliva) [inline CHALLENGE.md:21]: Simpler:

"- Completion duration ≈3 hours."

File: CHALLENGE.md [hunk 9]
```diff
@@ -2,22 +2,23 @@
...
+- Write the code you are usually writing. Avoid writing broken or sketchy code. We can't evaluate how good you are if the code is not good.
+- In case of any questions regarding the challenge, please, contact a member of the hiring team who shared this challenge with you.
 ## Introduction
 
-You are given a prepared SPA project that was bootstrapped with Create React App. The app must render a list of tournaments from a fake REST API (included in the project). New tournaments can be created and tournaments should be searchable (search functionality must be optimised with performance in mind), editable and deletable through the fake REST API. A theme and a few basic elements such as headings, button and input have already been created for you.
+You are given a prepared SPA project that was bootstrapped with Create React App. The app must render a list of tournaments from a fake REST API (included in the project). New tournaments can be created and tournaments should be searchable, editable and deletable through the fake REST API. A theme and a few basic elements such as headings, button and input have already been created for you.
```
Comment (i-oliva) [inline CHALLENGE.md:11]: This is a minimally structured Create React App SPA project.
The project includes:
- A theme.
- Basic components that are meant to be used by you. (If we don't mention the components here, we will see if the candidate is able to find them after we've told the candidate that there are some components for use).
- A fake REST API.

File: CHALLENGE.md (hunk 7 above)
Comment (w01fgang) [reply CHALLENGE.md:7]: I mentioned it explicitly because sometimes candidates say on the technical interview that they did the challenge sketchy and they usually write better code 😂

Comment (i-oliva) [reply CHALLENGE.md:7]: I honestly think it should be a given of the fact that completing a challenge that gives you a ticket to the company that you want to work for should be done with reasonable code... I don't think we as a company need to say it :sweat:

Comment (i-oliva) [reply CHALLENGE.md:7]: Hmm maybe it can be written that way? "We are expecting to see the best out of you for this challenge"

Comment (w01fgang) [reply CHALLENGE.md:7]: This point was already discussed, we don't want people to do more than we ask.
Maybe we need better wording for it.

File: CHALLENGE.md (hunk 4 above)
Comment (w01fgang) [reply CHALLENGE.md:51]: as we didn't define how do we want to handle error cases, then I'll leave it like this for now
maybe it's a good question for the tech interview
